pkg/virt-handler/devices/base.go

```diff
@@ -0,0 +1,16 @@
+package devices
+
+import (
+	"golang.org/x/sys/unix"
+
+	"kubevirt.io/kubevirt/pkg/virt-handler/isolation"
+)
+
+type Device interface {
+	// Right now, including bridge/veth, only setup is needed, since veths are deleted if namespaces where they are part of are deleted
```

Comment:
please test it. from what I saw, the veth part which is outside of the pod is not deleted.

Reply:
Her you go:

```bash
# Create two network namespaces
ip netns add 'mynamespace-1'
ip netns add 'mynamespace-2'

# Create a veth virtual-interface pair
ip link add 'myns-1-eth0' type veth peer name 'myns-2-eth0'

# Assign the interfaces to the namespaces
ip link set 'myns-1-eth0' netns 'mynamespace-1'
ip link set 'myns-2-eth0' netns 'mynamespace-2'

# List veth in both namespaces
ip netns exec 'mynamespace-1' ip a
ip netns exec 'mynamespace-2' ip a

# Delete second namespace
ip netns del 'mynamespace-2'

# List first namespace
ip netns exec 'mynamespace-1' ip a
```

Here the relevant output before deleting `mynamespace-2`

```bash
[root@localhost ~]# ip netns exec 'mynamespace-1' ip a
1: lo: <LOOPBACK> mtu 65536 qdisc noop state DOWN group default qlen 1000
    link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
1414: myns-1-eth0@if1413: <BROADCAST,MULTICAST> mtu 1500 qdisc noop state DOWN group default qlen 1000
    link/ether 96:a0:2e:60:18:5a brd ff:ff:ff:ff:ff:ff link-netnsid 1
[root@localhost ~]# ip netns exec 'mynamespace-2' ip a
1: lo: <LOOPBACK> mtu 65536 qdisc noop state DOWN group default qlen 1000
    link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
1413: myns-2-eth0@if1414: <BROADCAST,MULTICAST> mtu 1500 qdisc noop state DOWN group default qlen 1000
    link/ether 46:4e:25:91:af:7b brd ff:ff:ff:ff:ff:ff link-netnsid 0
```

And here `mynamespace-1` after `mynamespace-2` was deleted:

```bash
[root@localhost ~]# ip netns exec 'mynamespace-1' ip a
1: lo: <LOOPBACK> mtu 65536 qdisc noop state DOWN group default qlen 1000
    link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
```